File: example_topic_pkg/src/example_pub.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "cpp_pub_node");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Int32>("cpp_counter", 10);
    ros::Rate rate(2); // 2 Hz

    int count = 0;

    while (ros::ok())
    {
        std_msgs::Int32 msg;
        msg.data = count;
        pub.publish(msg);
        count++;
        rate.sleep();
    }

    return 0;
}

File: example_topic_pkg/scripts/example_pub.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Int32

rospy.init_node('py_pub_node')
pub = rospy.Publisher('py_counter', Int32, queue_size=10)
rate = rospy.Rate(2)

count = 0

while not rospy.is_shutdown():
    pub.publish(count)
    count += 1
    rate.sleep()
